File: packages/java/src/main/java/dev/kreuzberg/literllm/BudgetConfig.java
package dev.kreuzberg.literllm;

import java.util.Map;

/**
 * Configuration for cost budget enforcement.
 *
 * @param globalLimit
 *            maximum total cost allowed across all models, or {@code null} for
 *            no global limit
 * @param modelLimits
 *            per-model cost limits keyed by model name
 * @param enforcement
 *            enforcement mode: {@code "strict"} (reject requests exceeding
 *            budget) or {@code "warn"} (log a warning)
 */
public record BudgetConfig(Double globalLimit, Map<String, Double> modelLimits, String enforcement) {

	/**
	 * Creates a budget config with strict enforcement and no per-model limits.
	 *
	 * @param globalLimit
	 *            maximum total cost allowed
	 */
	public BudgetConfig(double globalLimit) {
		this(globalLimit, Map.of(), "strict");
	}
}

File: packages/csharp/LiterLlm/BudgetConfig.cs
namespace LiterLlm;

/// <summary>
/// Configuration for cost budget enforcement.
/// </summary>
/// <param name="GlobalLimit">
/// Maximum total cost allowed across all models, or <c>null</c> for no global limit.
/// </param>
/// <param name="ModelLimits">Per-model cost limits keyed by model name.</param>
/// <param name="Enforcement">
/// Enforcement mode: <c>"strict"</c> (reject requests exceeding budget)
/// or <c>"warn"</c> (log a warning).
/// </param>
public record BudgetConfig(
    double? GlobalLimit = null,
    IReadOnlyDictionary<string, double>? ModelLimits = null,
    string Enforcement = "strict");
